### PR TITLE
yarn v3

### PR DESCRIPTION
**Pull request by adonesky1**
Upgrades package manager from Yarn v1 to Yarn v3.

Followed the [migration guide](https://yarnpkg.com/getting-started/migration) + some of the example set in https://github.com/MetaMask/controllers/pull/846 for @metamask/controller

### CODE REVIEW

**Comment by adonesky1** [comment]
@mcmire @Gudahtt yarn v3 doesn't seem to like this thing I did where [I'm using a previous version of `@metamask/eth-hd-keyring`](https://github.com/MetaMask/eth-hd-keyring/commit/29710a190f017364e60d39f457bd59dde62c2858#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R42) in order to [demonstrate via test](https://github.com/MetaMask/eth-hd-keyring/commit/29710a190f017364e60d39f457bd59dde62c2858#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR30) that the new version derives the same accounts as the old when passed the same mnemonic. This is probably an anti-pattern and may have served its purpose already, but it's interesting that yarn v3 seems to disallow it. Should I just remove the test ya think?

**Comment by mcmire** [comment]
@adonesky1 It's likely that Yarn v3 is now seeing that the dependency in `package.json` refers to itself and ignores it.

What was the intent for this test? The version that you're linking to is 4.0.2 — but that's the version in the `package.json`. So as it stands, that isn't an old version. I think you'd need to publish a new version in order to be able to use an old version like that.

**Comment by Gudahtt** [comment]
Interesting that this used to work on Yarn v1! You can tell by the lockfile that it was getting it from the registry, despite it matching the local package version. Switching to v4.0.1 seems like a reasonable workaround.

File: .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable */
+//prettier-ignore
+module.exports = {
+    name: "@yarnpkg/plugin-allow-scripts",
```
Comment (Gudahtt) [inline .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs:4]: Unexpected indentation here

File: package.json [hunk 2]
```diff
@@ -62,7 +62,9 @@
     "allowScripts": {
       "@lavamoat/preinstall-always-fail": false,
       "keccak": false,
-      "secp256k1": false
+      "secp256k1": false,
+      "@metamask/eth-hd-keyring>ethereumjs-util>ethereum-cryptography>keccak": false,
```
Comment (Gudahtt) [inline package.json:66]: These are the updated paths for the two packages previously listed here. The old entries can be removed, they don't work anymore. The v2 release of allow-scripts updated how these package identifiers are generated, but it doesn't remove stale entries.

Comment (adonesky1) [reply package.json:64]: removed [here](https://github.com/MetaMask/eth-hd-keyring/pull/72/commits/5e79a0d1282bdeb87ebae83e443bd396d1c9ada3)

File: .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs (hunk 1 above)
Comment (adonesky1) [reply .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs]: fixed [here](https://github.com/MetaMask/eth-hd-keyring/pull/72/commits/5e79a0d1282bdeb87ebae83e443bd396d1c9ada3)

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**